src/sounds.cpp

```diff
@@ -572,9 +572,12 @@ void sfx::generate_gun_sound( const player &p, const item &firing )
         angle = 0;
         distance = 0;
         selected_sound = "fire_gun";
-        if( firing.has_gunmod( "suppressor" ) == 1 || firing.has_gunmod( "homemade suppressor" ) == 1 ) {
+
+        const auto mods = firing.gunmods();
+        if( std::any_of( mods.begin(), mods.end(), []( const item *e ) { return e->type->gunmod->loudness < 0; } ) {
```

Comment:
VERSION "0.C-11499-gad34e0d"

```
src/sounds.cpp: In lambda function:
src/sounds.cpp:577:88: error: invalid use of incomplete type 'const struct itype'
         if( std::any_of( mods.begin(), mods.end(), []( const item *e ) { return e->type->gunmod->loudness < 0; } ) {
                                                                                        ^
In file included from src/map.h:11:0,
                 from src/sounds.cpp:5:
src/item.h:24:8: note: forward declaration of 'struct itype'
 struct itype;
        ^
src/sounds.cpp: In function 'void sfx::generate_gun_sound(const player&, const item&)':
src/sounds.cpp:577:116: error: expected ')' before '{' token
         if( std::any_of( mods.begin(), mods.end(), []( const item *e ) { return e->type->gunmod->loudness < 0; } ) {
                                                                                                                    ^
src/sounds.cpp:581:5: error: expected primary-expression before '}' token
     } else {
     ^
In file included from C:/msys64/mingw64/include/c++/5.3.0/bits/stl_algobase.h:71:0,
                 from C:/msys64/mingw64/include/c++/5.3.0/bits/char_traits.h:39,
                 from C:/msys64/mingw64/include/c++/5.3.0/string:40,
                 from src/json.h:6,
                 from src/enums.h:7,
                 from src/sounds.h:4,
                 from src/sounds.cpp:1:

```

I still haven't been able to fully update my msys2 install (in particular several SDL packages) so if this is related to that, my apologies. I was however able to compile and run  #15631 , so I thought I would mention it.